File: frontend/src/types/room.ts
export interface RoomResponse {
  id: number;
  code: string;
  genre: string;
  status: string;
  created_at: string;
  member_count: number;
}

export interface RoomMembershipResponse extends RoomResponse {
  user_id: number;
}

export interface CreateRoomRequest {
  creator_name: string;
  genre: string;
}

export interface JoinRoomRequest {
  room_code: string;
  user_name: string;
}

File: backend/app/schemas/room.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CreateRoomRequest(BaseModel):
    creator_name: str
    genre: str


class JoinRoomRequest(BaseModel):
    room_code: str
    user_name: str


class RoomResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    code: str
    genre: str
    status: str
    created_at: datetime
    member_count: int


class RoomMembershipResponse(RoomResponse):
    """RoomResponse plus the calling user's own id — for endpoints where a
    specific user is doing the calling (create/join), unlike the plain
    room lookup which has no single "calling user"."""

    user_id: int
